server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -321,7 +360,8 @@ public ShardAllocationDecision decideShardAllocation(final ShardRouting shard, f
             threshold,
             preferPrimaryShardBalance,
             preferPrimaryShardRebalance,
-            ignoreThrottleInRestore
+            ignoreThrottleInRestore,
+            null // as we don't need to check if timed out or not while just understanding ShardAllocationDecision
```

Comment:
will prefer a no-op function that returns false instead of null to avoid running into NPE

Reply:
Ack, will update